sslyze/scanner.py

```diff
@@ -264,7 +277,7 @@ def get_results(self) -> Iterable[ServerScanResult]:
                         result = implementation_cls.result_for_completed_scan_jobs(
                             server_info, list(completed_scan_jobs)
                         )
-                        server_scan_results[scan_cmd] = result
+                        server_scan_results = dataclasses_replace(server_scan_results, scan_cmd=result)
```

Comment:
As I wanted to make the dataclass ``ScanCommandResults`` ``frozen`` as all other, we have to use ``replace`` and create a new Instance on every new completed scan job. As this is only a small container of other objects (which do not have to be copied/transferred), this should be ok (in performance).

The alternative would be to create a "unfrozen" version and then somehow "freeze" it.